conf/defaults.ini

```diff
@@ -768,7 +768,9 @@ icon = microsoft
 enabled = false
 allow_sign_up = true
 auto_login = false
+client_authentication = client_secret_post # either client_secret_post or client_secret_jwt
```

Comment:
**Re-commenting on current version of file for alignment with the latest version:**

@kalleep Update:
So I just tested the implementation of the oidc spec with client_secret_jwt when managed_identity_client_id is null, and I got this error:
ERROR[11-06|10:41:08] Failed to authenticate request           logger=authn.service client=auth.client.azuread error="[auth.oauth.token.exchange] failed to exchange code to token: oauth2: \"invalid_client\" \"AADSTS5002738: Invalid JWT token. 'HS256' is not a supported signature algorithm. Supported signing algorithms are: 'RS256,RS384,RS512,PS256,PS384,PS512' Trace ID: aa30e445-255c-49c3-b644-b10ddf269100 Correlation ID: ffa6b608-93ff-4af0-ace6-2b1161f512cf Timestamp: 2024-11-06 18:41:09Z\" \"https://login.microsoftonline.com/error?code=5002738\""

It seems as though Microsoft no longer supports HS256 (symmetric key encryption), and only supports RS and PS (asymmetric key encryption), most likely due to security concerns. I do have all three implementations coded and will wait for your feedback on this error.

Also - I just updated the PR so you can go ahead and review all current changes! These have been verified to work besides the case mentioned above

Reply:
Yes I can see that. So `client_secret_jwt` is not even an option.

In my mind managed identities was implemented using something like that but apparently not. `client_secret_jwt` is something we want to support if idp:s support it. Then I would suggest having managed identity as its own `client_authentication` option. This is one we can only implement for azure anyway.

So we can remove the related code for `client_secret_jwt`. And in exchange you would check if `managed_identity` is set in `client_authentication` we try to authenticate using that.

Reply:
@kalleep Sounds good! I'll go ahead and change it to `managed_identity` and delete the `client_secret_jwt` code.